nixos/modules/services/development/lorri.nix

```diff
@@ -0,0 +1,57 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.lorri;
+  socketUnit = "lorri";
+  socketPath = "lorri/daemon.socket";
+in with lib; {
+  options = {
+    services.lorri = {
+      enable = mkOption {
+        default = false;
+        type = types.bool;
+        description = ''
+          This option enables a systemd socket unit that listens on the
+          well-known address used by lorri, a nix-shell replacement for project
+          development. When a lorri client connects to the socket, systemd
+          starts the lorri daemon as a service.
```

Comment:
```
Enables the daemon for `lorri`, a nix-shell replacement for project development.
Socket-activated, daemon starts on first request by the `lorri` command.
```

Huh, cool, when we enable timeouts for the daemon, we could kill it when all projects time out and the socket activation would take care of restarting it.

Reply:
I'm not necessarily against your suggestion, but it would be good to know in what way you think it improves on what's currently there.

Reply:
Here’s my thinking:

The user needs the following information (in prioritized order):
.) what is enabled (the lorri daemon)
.) what lorri is (mostly for good grepp-ability in the manual, people searching for `nix-shell` for example)
.) That it’s socket-activated (optional what socket activation is).

You can leave out `This option` (because it’s trivially true), and if you reorder by the priority above it becomes what I suggested :)
Oh, you can also leave out `well-known address`, because that’s implied by socket activation.

Reply:
Changed, thanks for your explanation!